### PR TITLE
Integration tests migration

### PR DESCRIPTION
**Pull request by nikoo28**
Moved integration tests from `test-app` to the library itself.
Additional task to remove the `test-app` job and its contents will be done at a later stage.

Additional integration tests to verify cross partition encryption/decryption would be written once this gets merged in.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -191,6 +191,12 @@ jobs:
             source ./.circleci/set_env_variables.sh
             cd java/app-encryption
             ./scripts/test.sh
+      - run:
```
Comment (sushantmimani) [inline .circleci/config.yml:194]: Think it should be ok to remove the test-app step from circleci in this PR, even if the actual directory is removed later

File: java/app-encryption/src/it/java/com/godaddy/asherah/ConfigReader.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package com.godaddy.asherah;
+
+public class ConfigReader {
```
Comment (sushantmimani) [inline java/app-encryption/src/it/java/com/godaddy/asherah/ConfigReader.java:3]: Rename this class to something like `TestConfiguration` as we really aren't reading anything here

File: java/app-encryption/src/it/java/com/godaddy/asherah/ConfigReader.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  private String metastoreType;
+  private String metastoreJdbcUrl;
```
Comment (sushantmimani) [inline java/app-encryption/src/it/java/com/godaddy/asherah/ConfigReader.java:6]: Add params for username, password, connectionpool size (like the existing test-app) for more flexibility.

File: java/app-encryption/src/it/resources/config.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+metastoreType: "memory"
+metastoreJdbcUrl: ""
```
Comment (sushantmimani) [inline java/app-encryption/src/it/resources/config.yaml:2]: add params for username, password, connectionPoolSize for more flexible configuration

**Comment by nikoo28** [comment]
This closes https://github.com/godaddy/asherah/issues/247

**Pull request merged**